### PR TITLE
Added docker host detection and support for setting container host explicitly with env variable 

### PR DESCRIPTION
**Pull request by tuvshuud**
Changes:
- Use docker host address and mapped port if possible
- TC_HOST env to set docker host

Fix to #81, #74, #43

### CODE REVIEW

File: testcontainers/core/container.py [hunk 1]
```diff
@@ -77,17 +78,29 @@ def __del__(self):
                 pass
 
     def get_container_host_ip(self) -> str:
-        # if testcontainers itself runs in docker, get the newly spawned
-        # container's IP address from the dockder "bridge" network
-        if inside_container():
-            return self.get_docker_client().bridge_ip(self._container.id)
+        # https://github.com/testcontainers/testcontainers-go/blob/dd76d1e39c654433a3d80429690d07abcec04424/docker.go#L644
+        # if os env TC_HOST is set, use it
+        if os.environ.get('TC_HOST') is not None:
```
[the remark this answers is not in the document]
Comment (tuvshuud) [reply testcontainers/core/container.py]: You're right. In that case, we continue to get host address rather than failing silently.

**Comment by tillahoffmann** [comment]
Hey, I've merged your branch into the `gitlab2` branch (because we can't test on pull requests from forks on gitlab ci). Unfortunately, the tests seem to fail. See [here](https://gitlab.com/tillahoffmann/testcontainers-python/-/jobs/525235635) for the logs. 

Do you know why that might be happening? Quite possibly a result of the [config](https://github.com/testcontainers/testcontainers-python/blob/gitlab2/.gitlab-ci.yml) not being right.

**Comment by tuvshuud** [comment]
I have looked into failed test cases. They are required fix in this PR. I think they are used to fail in the docker:dind before. I will push some more commits to fix these when I have some time in the next week.

https://github.com/tuvshuud/testcontainers-python/blob/c058fd0d02ded23d21a9befb61dc4651155bfcb2/tests/test_docker_compose.py#L32

https://github.com/tuvshuud/testcontainers-python/blob/c058fd0d02ded23d21a9befb61dc4651155bfcb2/tests/test_new_docker_api.py#L35

**Comment by tuvshuud** [comment]
@tillahoffmann I have fixed the test cases and improved host detection code. Also, I ran all the test cases on Gitlab CI/CD and all passed. Can you merge into gitlab2 and check it now?

**Pull request merged**